### PR TITLE
Kdoc for datasetEditing fragments.

### PR DESCRIPTION
**Pull request by NicolasVial**
I added some KDoc but maybe some parts are not clear enough and also need some Kdoc to be fully understandable, tell me if you think so.

### CODE REVIEW

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureFragment.kt [hunk 1]
```diff
@@ -43,10 +47,10 @@ class AddPictureFragment : Fragment() {
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
+    ): View {
         parentActivity = requireActivity()
         _binding = FragmentAddPictureBinding.inflate(inflater, container, false)
-        return binding.root
+        return _binding!!.root
```
Comment (martin-lenweiter) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureFragment.kt]: Why is this necessary? Good that you changed it everywhere if so, then it's consistent.

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayImageSetFragment.kt [hunk 2]
```diff
@@ -26,7 +26,18 @@ import kotlinx.coroutines.launch
...
 
-
+/**
+ * Fragment to display all the pictures of a category.
+ *
+ * The name of the category and all the pictures of the category are displayed.
+ *
+ * Possible actions :
+ * - Get the information of what the user can do by clicking on the info menu button.
+ * - Can delete one or multiple pictures with a long click on one picture and then by clicking
```
Comment (martin-lenweiter) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayImageSetFragment.kt:35]: Nice to list all possible actions.

**Review by martin-lenweiter — COMMENTED** [review]
All good, just have a small question about the binding thing you can see in my comments, and then I'll approve

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureFragment.kt (hunk 1 above)
Comment (NicolasVial) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureFragment.kt]: hum to be honest I am not 100% sure to fully understand but if you want to undertsand, I think you should look at the Code review about the AddPictureFragment.

Comment (bbjubjub2494) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureFragment.kt]: short answer: `binding` is an alias for `_binding!!` and it comes from Google sample code somewhere.

Longer answer: it's so that the non-null assertion can be implicit and assume that it's good as long as the fragment hasn't been destroyed.

Remark: this change is wrong and should be reversed.

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayDatasetFragment.kt [hunk 3]
```diff
@@ -103,9 +116,13 @@ class DisplayDatasetFragment : Fragment() {
 
         lifecycleScope.launch {
             dataset = dbManagement.getDatasetById(datasetId)!!
-            (binding.displayDatasetName as TextView).text = dataset.name
+            (_binding?.displayDatasetName as TextView).text = dataset.name
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayDatasetFragment.kt]: This doesnt avoid the NPE
```suggestion
            (_binding?.displayDatasetName as? TextView)?.text = dataset.name
```

Comment (NicolasVial) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayDatasetFragment.kt]: you're totally right Thanks !

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureFragment.kt (hunk 1 above)
Comment (NicolasVial) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/AddPictureFragment.kt]: Okay ! I will change all my modifications and leave it as it was. Thanks you for the explanation !

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayImageSetFragment.kt [hunk 4]
```diff
@@ -282,6 +325,10 @@ class DisplayImageSetFragment : Fragment() {
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         return when (item.itemId) {
+            /**
+             * When the info menu button is clicked, display information to the user about
+             * the possible actions possible.
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayImageSetFragment.kt]:
```suggestion
             * the possible actions.
```

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/SelectPictureFragment.kt [hunk 5]
```diff
@@ -52,13 +54,22 @@ class SelectPictureFragment : Fragment() {
         categories = categories.plus(args.categories.toList())
 
         binding.choosePictureButton.setOnClickListener {
+            /**
+             * Allow the user to select an existing picture in it's device.
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/SelectPictureFragment.kt]:
```suggestion
             * Allow the user to select an existing picture in their device.
```

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]
couple typos, otherwise looks very good

**Review by bbjubjub2494 — APPROVED** [review]
LGTM. please solve the conflicts.

**Review by martin-lenweiter — APPROVED** [review]
All Good!
